From 897da4c270f262afc72109642c2fb7c630424e37 Mon Sep 17 00:00:00 2001
From: Marius Bagassien <marius.b@variscite.com>
Date: Wed, 18 Feb 2026 16:19:54 +0100
Subject: [PATCH] Android: Changes required to compile SCFW

Add SCFW compilation step for VAR-SOM-MX8X (som_mx8qx) and
VAR-SOM-MX8Q (som_mx8qm) products.

Based on Android 13 patches adapted for Android 15.

Signed-off-by: Marius Bagassien <marius.b@variscite.com>
---
 common/tools/imx-make.sh | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/common/tools/imx-make.sh b/common/tools/imx-make.sh
index c95f2631..f78e03c5 100755
--- a/common/tools/imx-make.sh
+++ b/common/tools/imx-make.sh
@@ -118,6 +118,10 @@ for arg in ${args[*]} ; do
     esac
 done
 
+if [ "${TARGET_PRODUCT}" = "som_mx8qx" ] || [ "${TARGET_PRODUCT}" = "som_mx8qm" ]; then
+	source $ANDROID_BUILD_TOP/device/variscite/imx8q/som_mx8q/uboot-firmware/scfw_build.sh ${TARGET_PRODUCT}
+fi
+
 # if bootloader and kernel not in arguments, all need to be made
 if [ "${build_bootloader}" = "" ] && [ "${build_kernel}" = "" ] && \
         [ "${build_kernel_modules}" = "" ] && [ "${build_kernel_dts}" = "" ] && \
-- 
2.43.0

